tests/Agent/IntegrationTests/UnboundedIntegrationTests/Elasticsearch/ElasticsearchTests.cs

```diff
@@ -0,0 +1,324 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using MultiFunctionApplicationHelpers;
+using NewRelic.Agent.IntegrationTestHelpers;
+using NewRelic.Agent.IntegrationTests.Shared;
+using NewRelic.Testing.Assertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NewRelic.Agent.UnboundedIntegrationTests.Elasticsearch
+{
+    public abstract class ElasticsearchTestsBase<TFixture> : NewRelicIntegrationTest<TFixture>
+        where TFixture : ConsoleDynamicMethodFixture
+    {
+        protected enum ClientType
+        {
+            ElasticsearchNet,
+            NEST,
+            ElasticClients
+        }
+
+        protected readonly ConsoleDynamicMethodFixture _fixture;
+
+        protected readonly string _host = GetHostFromElasticServer(ElasticSearchConfiguration.ElasticServer);
+
+        protected readonly ClientType _clientType;
+
+        const string IndexName = "flights";
+
+
+        protected ElasticsearchTestsBase(TFixture fixture, ITestOutputHelper output, ClientType clientType) : base(fixture)
+        {
+            _fixture = fixture;
+            _fixture.TestLogger = output;
+            _clientType = clientType;
+
+            // TODO: Set high to allow for debugging
+            _fixture.SetTimeout(TimeSpan.FromMinutes(20));
+
+            _fixture.AddCommand($"ElasticsearchExerciser SetClient {clientType}");
+
+            // Sync operations
+
+            _fixture.AddCommand($"ElasticsearchExerciser Index");
+
+            _fixture.AddCommand($"ElasticsearchExerciser Search");
+
+            _fixture.AddCommand($"ElasticsearchExerciser IndexMany");
+
+            _fixture.AddCommand($"ElasticsearchExerciser MultiSearch");
+
+            // Async operations
+
+            _fixture.AddCommand($"ElasticsearchExerciser IndexAsync");
+
+            _fixture.AddCommand($"ElasticsearchExerciser SearchAsync");
+
+            _fixture.AddCommand($"ElasticsearchExerciser IndexManyAsync");
+
+            _fixture.AddCommand($"ElasticsearchExerciser MultiSearchAsync");
+
+            // Don't like having to do this but it makes the async tests pass reliably
+            _fixture.AddCommand("RootCommands DelaySeconds 5");
```

Comment:
If you execute the async commands before the sync commands, does that also work consistently?

Reply:
That's a great idea, and I tried it.  Unfortunately it looks like the five second delay is necessary after the async commands for the tests to pass reliably even if we do the async ones first.